draft-ietf-quic-multipath.md

```diff
@@ -1416,16 +1416,23 @@ PATH_CIDS_BLOCKED frames are formatted as shown in {{fig-path-cid-blocked-frame-
 PATH_CIDS_BLOCKED Frame {
   Type (i) = TBD-09 (experiments use 0x15228c0e),
   Path Identifier (i),
+  Max Sequence Number (i),
 }
 ~~~
 {: #fig-path-cid-blocked-frame-format title="PATH_CIDS_BLOCKED Frame Format"}
 
 Path Identifier:
 : Identifier of the path for which unused connection IDs are not available.
 
+Maximum Sequence Number:
+: The largest sequence number issued for a connection ID for this path by the peer.
+
```

Comment:
What if the peer has not yet sent any CID?

Reply:
Okay, we would make this the next expected CID sequence number. So effectively the largest+1 or 0 if no CID was issued yet?

Reply:
Yes. But then, what is supposed to happen if the peer receives a fanciful number?

Reply:
Notice that having a sequence number does not by itself solve the repetition problem. Suppose the peer has not sent a CID yet. I observe that the packet has not been acked. Should I repeat the frame? What if this is a spurious repeat. Will I repeat the spurious repeat as long as the number has not increased?

In the end, I think I will have to keep the workaround that I had in the code -- a flag that says "there is an unacked CID blocked frame for this path", cleared when the frame is acked or when new CIDs arrive. But if I have that flag, the number of CID in the frame is just informational. The value is mostly to be nice with the receiver, as in "if the number expected is lower than what you have sent, don't bother." Kinda nice to have.

Reply:
OK. I think this is overall a good change, because it aligns the syntax of `PATH_CIDS_BLOCKED` with that of the very similar `STREAM_DATA_BLOCKED`. It helps processing at the receivers: they can ignore `PATH_CIDS_BLOCKED` if they have already sent more CIDs for this path than indicated by the sequence number.

The change does not completely solve the retransmission problem. It helps a bit -- no point repeating if the peer has already sent more CIDs. Endpoints still need to track acknowledgements if they want to avoid spurious retransmissions -- but then, that's true of every frame, so there may not be much need to say that again.